doc/guide/Guide_continuous_integration_GitHub_actions.ipynb

```diff
@@ -0,0 +1,137 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Continuous Integration and GitHub Actions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Automated Tests\n",
+    "\n",
+    "On Jenkins tests are executed and analyzed automatically, in an unbiased environment. The results are stored and can be compared with previous test runs.\\\n",
+    "Jenkins has a GUI for monitoring individual tests, full test runs and test result trands.\\\n",
+    "Developers are requested to watch it. At first when they push commits to the code repository, but also later on, when other changes in data or sources may make it necessary to review and refactor code that once passed all tests.\n",
+    "The CLIMADA Jenkins server used for continuous integration is at [(https://ied-wcr-jenkins.ethz.ch) ](https://ied-wcr-jenkins.ethz.ch).\n",
+    "\n",
+    "##### Developer guidelines:\n",
+    "\n",
+    "- All tests must pass before submitting a pull request.\n",
+    "- Integration tests don't run on feature branches in Jenkins, therefore developers are requested to run them locally.\n",
+    "- After a pull request was accepted and the changes are merged to the develop branch, integration tests may still fail there and have to be addressed."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Test Coverage\n",
+    "\n",
+    "Jenkins also has an interface for exploring code coverage analysis result.\\\n",
+    "This shows which part of the code has never been run in any test, by module, by function/method and even by single line of code.\n",
+    "\n",
+    "__Ultimately every single line of code should be tested.__\n",
+    "\n",
+    "##### Developer guidelines:\n",
+    "\n",
+    "- Make sure the coverage of novel code is at 100% before submitting a pull request.\n",
+    "\n",
+    "Be aware that the having a code coverage alone does not grant that all required tests have been written!\\\n",
+    "The following artificial example would have a 100% coverage and still obviously misses a test for `y(False)`"
```

Comment:
Which artificial example? I think it got lost.

Reply:
I lost a code block when moving this part of the guide around, I have added it back in now